pkg/providers/snow/defaults.go

```diff
@@ -0,0 +1,73 @@
+package snow
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+type Defaulters struct {
+	awsClientMap AwsClientMap
+	writer       filewriter.FileWriter
+}
+
+func NewDefaulters(aws aws.Clients, writer filewriter.FileWriter) *Defaulters {
+	return &Defaulters{
+		awsClientMap: NewAwsClientMap(aws),
+		writer:       writer,
+	}
+}
+
+func NewDefaultersFromAwsClientMap(awsClientMap AwsClientMap, writer filewriter.FileWriter) *Defaulters {
+	return &Defaulters{
+		awsClientMap: awsClientMap,
+	}
+}
+
+func (d *Defaulters) GenerateDefaultSshKey(ctx context.Context, m *v1alpha1.SnowMachineConfig) (keyName, keyVal string, err error) {
+	if m.Spec.SshKeyName != "" {
+		return "", "", nil
+	}
+
```

Comment:
nit: we probably can remove this?
```suggestion
```

Reply:
i kept it because `GenerateDefaultSshKey` is a exported method and I wanna prevent nil pointer when misused